sql/mysql/migrate.go

```diff
@@ -307,8 +309,17 @@ func (s *state) modifyTable(modify *schema.ModifyTable) error {
 	}
 	for i := range changes {
 		if len(changes[i]) > 0 {
-			if err := s.alterTable(modify.T, changes[i]); err != nil {
-				return err
+			// for TiDB - break down each alter table statement because it doesn't support multischema see: https://github.com/pingcap/tidb/issues/14766
+			if strings.Contains(s.version, "TiDB") {
+				for _, c := range changes[i] {
+					if err := s.alterTable(modify.T, []schema.Change{c}); err != nil {
+						return err
+					}
+				}
+			} else {
+				if err := s.alterTable(modify.T, changes[i]); err != nil {
+					return err
+				}
```

Comment:
This code can be deleted by wrapping the differ and breaking one ModifyTable command into n ModifyTable commands.